conan/tools/microsoft/msbuilddeps.py

```diff
@@ -284,4 +295,28 @@ def _content(self):
             dep_content = self._dep_props_file(dep_name, file_dep_name, props_name, condition)
             result[file_dep_name] = dep_content
 
+        build_req = list(self._conanfile.dependencies.build.values())
+        for dep in build_req:
+            dep_name = dep.ref.name
+            dep_name = dep_name.replace(".", "_") + "_build"
+            cpp_info = DepCppInfo(dep.cpp_info)  # To account for automatic component aggregation
```

Comment:
We could start using the `new_cpp_info` here and calling `aggregate_components` for that, the same for the `host/test`

Reply:
Agree, lets try it. 
However, we should probably refactor that, so we can pick a name for ``new_cpp_info`` that will survive in the transition to 2.0